redis.conf

```diff
@@ -1951,13 +1951,20 @@ list-max-listpack-size -2
 # etc.
 list-compress-depth 0
 
-# Sets have a special encoding in just one case: when a set is composed
+# Sets have a special encoding when a set is composed
 # of just strings that happen to be integers in radix 10 in the range
 # of 64 bit signed integers.
 # The following configuration setting sets the limit in the size of the
 # set in order to use this special memory saving encoding.
 set-max-intset-entries 512
 
+# Sets containing non-integer values are also encoded using a memory efficient
+# data structure when they have a small number of entries, and the biggest entry
+# does not exceed a given threshold. These thresholds can be configured using
+# the following directives.
+set-max-listpack-entries 128
```

Comment:
I picked the same value as for `zset-max-listpack-entries`. Didn't really analyze it but you're right, we should at least double that.

I'll make it 1024 if you think it's the best value. What speaks for a lower value is efficiency, unless those points are solved.

Reply:
interesting, it's odd that hash max entries is 4 times bigger than zset.
maybe it's because the zset's listpacks are sorted, unlike the hash ones?

i would argue that here was can use double of that, but then question is double of what?
double of zset or double of hash?
for now, let's take the middle ground, and use 512 (double of zset and same as hash).
any further decision will probably need to be based on the typical type of operations done on sets, and some benchmarks.